### PR TITLE
Updated Search CRD

### PR DESCRIPTION
**Pull request by xrajesh**

### CODE REVIEW

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:2]:
```
// Copyright Contributors to the Open Cluster Management project
```

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:39]: Does this references a ConfigMap? Referencing another object adds complexity. Why don't we define the required data here? 
Maybe the name could be `ExternalDBInstance`.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:56]: Maybe the name could be `ExternalDBInstance`.
What's the expected format for the secret?
Typically certs are managed by a cert service (ie. certmanager) so adding additional fields might be difficult.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:134]: 1. Need to confirm that we want to use labels vs ClusterSets.
2. If using labels, we should implement as a [LabelSelector](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1178-L1191) for consistency with K8s. It also adds more flexibility.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:131]: This needs to be a list of []ApiGroups and []Resources.
Example:
```yaml
  AllowedResources: 
  - apiGroups: [""]
    resources: ["pods", "alertmanager"]
  - apiGroups: ["open-cluster-management.io", "tuned.openshift.io"]
    resources: ["*"]
  DeniedResources:
  - apiGroups: [""]
    resources: ["secrets"]
  - apiGroups: ["open-cluster-management.io"]
    resources: ["policies", "iampolicies", "certificatepolicies"]
```

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:142]: `Search` is redundant, I'd name this just `Health`.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:147]: Maybe `InUse` is implied and not needed because status represents current state.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:162]: Should use the generic `Condition` from the Kubernetes API? https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:148]: Is it possible to define the array inline like this
```go
Conditions []SearchCondition `json:"conditions,omitempty"`
```

File: bundle/manifests/search-v2-operator.clusterserviceversion.yaml
Comment (jlpadilla) [inline bundle/manifests/search-v2-operator.clusterserviceversion.yaml:141]: Revert before merge.

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [inline api/v1alpha1/search_types.go:33]: @dockerymick Could you help review the code comments in this file? These comments become the API documentation.

Comment (dockerymick) [reply api/v1alpha1/search_types.go:33]: Yes! Will review shortly

File: api/v1alpha1/search_types.go
Comment (xrajesh) [reply api/v1alpha1/search_types.go:39]: This is name of the Configmap , isnt it like name of the storageClass @jlpadilla

File: api/v1alpha1/search_types.go
Comment (xrajesh) [reply api/v1alpha1/search_types.go:56]: This secret can like following 
```
data:
        db_url:
        db_name: 
        db_user:
        db_password:
        ca_cert:
```
For example the db can be in a VM , the certs will be provided by the customer right. @jlpadilla

Comment (jlpadilla) [reply api/v1alpha1/search_types.go:56]: - If postgres is on a VM, yes, but most likely there will be some automation to move the certs if these need to be rotated periodically.
- I'm thinking of the scenario where postgres is on the cluster, for example, ACM installs a postgres instance for the product and search uses the ACM postgres instead of our own.

Users will need to know the structure of this secret. Can we add something like this?
```
// ... the secret must contain these keys [db_host, db_port, db_user, db_password, db_name, ca_cert]
```

`db_url` is a compressed format like: `postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}`

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [reply api/v1alpha1/search_types.go:162]: What's the concern with upgrade?  If the kube API changes it will need to be published as `v2`, so it won't break us.

My concern is consistency. This is a concept that kubernetes has standardized across resources, so I don't want us to drift.

File: api/v1alpha1/search_types.go
Comment (xrajesh) [reply api/v1alpha1/search_types.go:25]: Yes Options 2 - is what we want - But creating a configmap from Key,value pair is easy - But this comes in only when we have to tune the DB.

File: api/v1alpha1/search_types.go
Comment (xrajesh) [reply api/v1alpha1/search_types.go:42]: This is mainly for ExternalDB - If ACM has a common Postgres instance
(1)  we will update our code to use that secret .

File: api/v1alpha1/search_types.go
Comment (jlpadilla) [reply api/v1alpha1/search_types.go:25]: Creating the configmap is easy, but it becomes another step and another object to track, backup, etc.

**Review by jlpadilla — APPROVED** [review]
/lgtm

Approving with a couple concerns with the API, but this is great progress!

File: README.md
Comment (dockerymick) [inline README.md:84]: Similar to the other comments. remove the number out of the section title

**Review by dockerymick — COMMENTED** [review]
@xrajesh @jlpadilla

File: PROJECT
Comment (jlpadilla) [reply PROJECT:17]: This is dev preview and not included in the product yet, but let's discuss how we can start creating this content to be merged when we GA in 2.6.

**Pull request merged**
